modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/ItDateTimeCastFormatTest.java

```diff
@@ -218,29 +216,27 @@ public void timeUpdateFromLiteral(DateTimeArgs<LocalTime> args) {
 
         if (args.value != null) {
             assertQuery("SELECT time0_col FROM datetime_cols WHERE id = 1")
-                    .returns(args.value)
+                    // We are writing to time0_col, so we should expect a time w/o a fractional part.
```

Comment:
Let's create a ticket to support precisions greater than 3 and add a TODO here, 
to not forget revisiting these tests when the issue will be fixes.

Also, truncating arguments in generator method  `@MethodSource("time")` may be more convenient. WDYT?

Reply:
> Let's create a ticket to support precisions greater than 3

We have https://issues.apache.org/jira/browse/IGNITE-19162 and several TODOs in our code base with it.

> add a TODO here
It's not relevant to sub-millis support, we have a column TIME(0) but updating it to a value with milliseconds. After this patch millis are truncated from the result value.

> truncating arguments in generator method @MethodSource("time") may be more convenient.
Agreed, but the arguments from the "time" method are also used for other tests that expect milliseconds as a result :pensive: 
The good solution - merge args from "time" with "timeWIthPrecision" and remove such excessive tests, but I'm too lazy... but if you insist, I'll try to combine them :pensive: